### PR TITLE
Fixes - 18539 - compile under windows

### PR DESCRIPTION
**Pull request by avril-gh**
windows has been detected as NOT windows

### CODE REVIEW

**Comment by avril-gh** [comment]
fixes #18539

File: modules/openssl/SCsub [hunk 1]
```diff
@@ -671,7 +671,7 @@ if (env['builtin_openssl'] != 'no'):
     # Workaround for compilation error with GCC/Clang when -Werror is too greedy (GH-4517)
     import os
     import methods
-    if not (os.name == "nt" and os.getenv("VSINSTALLDIR") != None):  # not Windows and not MSVC
+    if (os.name != "nt" and os.getenv("VSINSTALLDIR") == None):  # not Windows and not MSVC
```
Comment (akien-mga) [inline modules/openssl/SCsub]: The right fix would be to replace `VSINSTALLDIR` by `VCINSTALLDIR` a priori. Your fix will break compilation with MinGW on `nt`.

Comment (avril-gh) [reply modules/openssl/SCsub]: came to me. you'v been faster to spot it than i RE-PR  :smiley:
fixed, rebased and tested again. Works ok. :thumbsup:

**Comment by avril-gh** [comment]
changed thanks to @akien-mga suggestion, Rebased and tested again. Works ok

File: modules/openssl/SCsub (hunk 1 above)
Comment (neikeq) [reply modules/openssl/SCsub]: Shouldn't this be `env.msvc` instead of `os.getenv("VCINSTALLDIR") != None` after #17194?

Comment (akien-mga) [reply modules/openssl/SCsub]: Yeah but that's the 2.1 branch, where #17194 was not backported.

**Pull request merged**